---
layout: default
title: Research
permalink: /research/
image: research-banner.png
---


<div class="heroresearchbanner flex-wrapper flex-center" style="background-image: url('{{ site.baseurl }}/images/{{ page.image }}');">
</div>

<div class="entry">
<h2>Alternative splicing in single cells</h2>
<p>Alternative pre-mRNA splicing is a tightly regulated post-transcriptional process that amplifies the 
coding potential of the genome by co-expressing multiple RNA variants from the same gene. Its investigation at 
the level of single cells have been challenging thus far due to the limitations of single-cell sequencing 
technology and current alternative splicing tools have inaccurately reported that most multi-exonic 
genes tend to express a single isoform at a time. We aim to develop new computational tools using 
novel statistical approaches to accurately quantify levels of exon splicing and implement it to study 
changes in alternative splicing during neuronal development.</p>

<h2>Splicing regulation using network theory</h2>
<p>The outcome of alternative splicing is tightly regulated by the activity of RNA-binding protein
modules that bind onto regions close to an alternative exon. These regulators of splicing may exert
opposing effects on splicing decisions and some of these proteins play different functions depending
on its binding position. We aim to elucidate the underlying regulation that governs alternative splicing 
decisions by developing a suite of bioinformatics tools that constructs a regulatory network of
RNA-binding proteins and its influence on the splicing of alternative exons.</p>

<h2>Heterogeneity of nonsense-mediated decay (NMD) activity</h2>
<p>NMD is a highly conserved quality control mechanism that enforces the accuracy of gene expression by 
clearing transcripts harboring premature termination codons. The success of many biotechnology and 
biomedical applications such as CRISPR-Cas9 gene knockout systems and cancer immunotherapies rely 
on optimal NMD activity. The efficacy of NMD is highly variable between biological systems and between
individual cells but little is known about the underlying mechanism that influences this decay pathway.
We aim to achieve broader understanding of the predictors of 
NMD activity using advanced statistical and machine learning models.</p>

</div>
