<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Expression/samples-related functions — factR-exp • factR2</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Expression/samples-related functions — factR-exp"><meta property="og:description" content="A set of functions to incorporate expression data into factR object and
tests regulatory potential of alternative splicing events.
Correlates exon inclusion levels with gene expression levels"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">factR2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/factR2.html">
    <span class="fas fa-book-open"></span>
     
    Using factR2
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper"></span>
     
    News
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="..//index.html">
    <span class="fas fa-globe"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/f-hamidlab/factR2" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/FurshamH" class="external-link">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Expression/samples-related functions</h1>
    
    <div class="hidden name"><code>factR-exp.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A set of functions to incorporate expression data into factR object and
tests regulatory potential of alternative splicing events.</p>
<p>Correlates exon inclusion levels with gene expression levels</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="co"># S4 method for factR</span></span>
<span><span class="fu">addTxCounts</span><span class="op">(</span><span class="va">object</span>, <span class="va">countData</span>, sampleData <span class="op">=</span> <span class="cn">NULL</span>, psi <span class="op">=</span> <span class="cn">NULL</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for factR</span></span>
<span><span class="fu">testGeneCorr</span><span class="op">(</span><span class="va">object</span>, vst <span class="op">=</span> <span class="cn">TRUE</span>, min_n <span class="op">=</span> <span class="fl">3</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>object</dt>
<dd><p>factR object</p></dd>


<dt>countData</dt>
<dd><p>Can be one of the following:</p><ul><li><p>Path to local count matrix file in .tsv or .csv format</p></li>
<li><p>Matrix object containing transcript-level expression counts data.</p></li>
</ul></dd>


<dt>sampleData</dt>
<dd><p>(Optional) Dataframe containing samples information. Dataframe rows
can be named to match the sample names in `countData`. If `sampleData` has no
row names, function will attempt to pick the column containing sample names
and assign it as rownames.</p></dd>


<dt>psi</dt>
<dd><p>(Optional) Exon-level splicing inclusion data. Can be one of the following:</p><ul><li><p>Path to local file in .tsv or .csv format</p></li>
<li><p>Matrix object</p></li>
</ul><p>Rownames of matrix should be in chr:start-end format</p></dd>


<dt>verbose</dt>
<dd><p>Boolean value as to whether messages should be printed (Default: TRUE)</p></dd>


<dt>vst</dt>
<dd><p>whether to apply variance stabilization on splicing and expression levels</p></dd>


<dt>min_n</dt>
<dd><p>minimum number of non-NA samples required for correlation testing</p></dd>


<dt>...</dt>
<dd><p>additional arguments parsed to cor.test function</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>factRObject with updated counts data and samples metadata.</p>


<p>factRObject with updated ASE metadata</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>`addTxCounts` converts transcript-level expression counts into gene-level
expression counts. If "psiData" is not provided, this function will determine
the inclusion level of alternative splicing events (psi) for each sample
by calculating the proportion of transcripts containing the splicing event
over all transcripts that spans that splicing event. If "psiData" is provided,
will check for common exon coordinates and non-overlapping ones detected
by factR2 will be set to NA. Row names of "sampleData" dataframe can be unnamed,
and this function will search for column variable corresponding to sample names.
If unsuccessful or if row names of "sampleData" does not match to column names
of "countData", an error will be returned.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="factRObject-class.html">factRObject-class</a></code> <code><a href="factR-exp-meta.html">factR-exp-meta</a></code></p>
<p><code><a href="https://rdrr.io/r/stats/cor.test.html" class="external-link">cor.test</a></code> <code>factR-exp</code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## get path to sample GTF and expression data</span></span></span>
<span class="r-in"><span><span class="va">gtf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"pb_custom.gtf.gz"</span>, package <span class="op">=</span> <span class="st">"factR2"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"pb_expression.tsv.gz"</span>, package <span class="op">=</span> <span class="st">"factR2"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## create factRObject with expression counts</span></span></span>
<span class="r-in"><span><span class="va">factRObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="createfactRObject.html">createfactRObject</a></span><span class="op">(</span><span class="va">gtf</span>, <span class="st">"vM33"</span>, countData <span class="op">=</span> <span class="va">counts</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00; font-weight: bold;">🡆  Checking inputs</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00; font-weight: bold;">🡆  Checking factRObject</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00; font-weight: bold;">🡆  Adding custom transcriptome</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">    ℹ </span><span style="color: #0000BB; font-style: italic;">Importing from local directory</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00; font-weight: bold;">🡆  Adding annotation</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">    ℹ </span><span style="color: #0000BB; font-style: italic;">Importing from URL</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00; font-weight: bold;">🡆  Adding genome sequence</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">    ℹ </span><span style="color: #0000BB; font-style: italic;">Using BSgenome object</span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>4 seqlevel(s) in ``your custom GTF`` are not found in ``reference annotation``</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00; font-weight: bold;">🡆  Matching chromosome names</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00; font-weight: bold;">🡆  Matching gene information</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-style: italic;">    Number of mismatched gene_ids found: 83520</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-style: italic;">    --&gt; Attempting to match ensembl gene_ids...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-style: italic;">    --&gt; No ensembl gene ids found in query</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-style: italic;">    ---&gt; Attempting to match gene_ids by finding overlapping coordinates...</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-style: italic;">    ---&gt; 65203 gene_id matched</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-style: italic;">    Total gene_ids corrected: 65203</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="font-style: italic;">    Remaining number of mismatched gene_ids: 18317</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00; font-weight: bold;">🡆  Creating factRset objects</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">    ℹ </span><span style="color: #0000BB; font-style: italic;">Adding gene information</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">    ℹ </span><span style="color: #0000BB; font-style: italic;">Adding transcript information</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">    ℹ </span><span style="color: #0000BB; font-style: italic;">Adding alternative splicing information</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00; font-weight: bold;">🡆  Annotating novel transcripts</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00; font-weight: bold;">🡆  Annotating novel AS events</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00; font-weight: bold;">🡆  Adding expression data</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">    ℹ </span><span style="color: #0000BB; font-style: italic;">Importing from local file</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00; font-weight: bold;">🡆  Creating samples metadata</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00; font-weight: bold;">🡆  Processing expression data</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">    ℹ </span><span style="color: #0000BB; font-style: italic;">Adding gene counts</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">    ℹ </span><span style="color: #0000BB; font-style: italic;">Adding spliced-event counts</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">    ℹ </span><span style="color: #0000BB; font-style: italic;">Normalizing counts</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ </span><span style="color: #0000BB; font-style: italic;">factRobject created!</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### This can also be done post-creation of a factR object</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="va">factRObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="createfactRObject.html">createfactRObject</a></span><span class="op">(</span><span class="va">gtf</span>, <span class="st">"vM33"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">factRObject</span> <span class="op">&lt;-</span> <span class="fu">addTxCounts</span><span class="op">(</span><span class="va">factRObject</span>, <span class="va">counts</span><span class="op">)</span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## access counts data</span></span></span>
<span class="r-in"><span><span class="fu">counts</span><span class="op">(</span><span class="va">factRObject</span><span class="op">)</span>  <span class="co"># returns normalised expression data from current Set</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in (function (classes, fdef, mtable) {    methods &lt;- .findInheritedMethods(classes, fdef, mtable)    if (length(methods) == 1L)         return(methods[[1L]])    else if (length(methods) == 0L) {        cnames &lt;- paste0("\"", vapply(classes, as.character,             ""), "\"", collapse = ", ")        stop(gettextf("unable to find an inherited method for function %s for signature %s",             sQuote(fdef@generic), sQuote(cnames)), domain = NA)    }    else stop("Internal error in finding inherited methods; didn't return a unique method",         domain = NA)})(list(structure("factR", package = "factR2")), new("standardGeneric",     .Data = function (object, ...)     standardGeneric("counts"), generic = structure("counts", package = "BiocGenerics"),     package = "BiocGenerics", group = list(), valueClass = character(0),     signature = "object", default = NULL, skeleton = (function (object,         ...)     stop(gettextf("invalid call in method dispatch to '%s' (no default method)",         "counts"), domain = NA))(object, ...)), &lt;environment&gt;):</span> unable to find an inherited method for function ‘counts’ for signature ‘"factR"’</span>
<span class="r-in"><span><span class="fu">counts</span><span class="op">(</span><span class="va">factRObject</span>, <span class="st">"Ptbp1"</span><span class="op">)</span>  <span class="co"># same as above, but only for specific genes</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in (function (classes, fdef, mtable) {    methods &lt;- .findInheritedMethods(classes, fdef, mtable)    if (length(methods) == 1L)         return(methods[[1L]])    else if (length(methods) == 0L) {        cnames &lt;- paste0("\"", vapply(classes, as.character,             ""), "\"", collapse = ", ")        stop(gettextf("unable to find an inherited method for function %s for signature %s",             sQuote(fdef@generic), sQuote(cnames)), domain = NA)    }    else stop("Internal error in finding inherited methods; didn't return a unique method",         domain = NA)})(list(structure("factR", package = "factR2")), new("standardGeneric",     .Data = function (object, ...)     standardGeneric("counts"), generic = structure("counts", package = "BiocGenerics"),     package = "BiocGenerics", group = list(), valueClass = character(0),     signature = "object", default = NULL, skeleton = (function (object,         ...)     stop(gettextf("invalid call in method dispatch to '%s' (no default method)",         "counts"), domain = NA))(object, ...)), &lt;environment&gt;):</span> unable to find an inherited method for function ‘counts’ for signature ‘"factR"’</span>
<span class="r-in"><span><span class="fu">counts</span><span class="op">(</span><span class="va">factRObject</span>, <span class="st">"Ptbp1"</span>, set <span class="op">=</span> <span class="st">"transcript"</span><span class="op">)</span>  <span class="co"># get transcript-level expression</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in (function (classes, fdef, mtable) {    methods &lt;- .findInheritedMethods(classes, fdef, mtable)    if (length(methods) == 1L)         return(methods[[1L]])    else if (length(methods) == 0L) {        cnames &lt;- paste0("\"", vapply(classes, as.character,             ""), "\"", collapse = ", ")        stop(gettextf("unable to find an inherited method for function %s for signature %s",             sQuote(fdef@generic), sQuote(cnames)), domain = NA)    }    else stop("Internal error in finding inherited methods; didn't return a unique method",         domain = NA)})(list(structure("factR", package = "factR2")), new("standardGeneric",     .Data = function (object, ...)     standardGeneric("counts"), generic = structure("counts", package = "BiocGenerics"),     package = "BiocGenerics", group = list(), valueClass = character(0),     signature = "object", default = NULL, skeleton = (function (object,         ...)     stop(gettextf("invalid call in method dispatch to '%s' (no default method)",         "counts"), domain = NA))(object, ...)), &lt;environment&gt;):</span> unable to find an inherited method for function ‘counts’ for signature ‘"factR"’</span>
<span class="r-in"><span><span class="fu">counts</span><span class="op">(</span><span class="va">factRObject</span>, <span class="st">"Ptbp1"</span>, set <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span> <span class="co"># get gene-level expression</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in (function (classes, fdef, mtable) {    methods &lt;- .findInheritedMethods(classes, fdef, mtable)    if (length(methods) == 1L)         return(methods[[1L]])    else if (length(methods) == 0L) {        cnames &lt;- paste0("\"", vapply(classes, as.character,             ""), "\"", collapse = ", ")        stop(gettextf("unable to find an inherited method for function %s for signature %s",             sQuote(fdef@generic), sQuote(cnames)), domain = NA)    }    else stop("Internal error in finding inherited methods; didn't return a unique method",         domain = NA)})(list(structure("factR", package = "factR2")), new("standardGeneric",     .Data = function (object, ...)     standardGeneric("counts"), generic = structure("counts", package = "BiocGenerics"),     package = "BiocGenerics", group = list(), valueClass = character(0),     signature = "object", default = NULL, skeleton = (function (object,         ...)     stop(gettextf("invalid call in method dispatch to '%s' (no default method)",         "counts"), domain = NA))(object, ...)), &lt;environment&gt;):</span> unable to find an inherited method for function ‘counts’ for signature ‘"factR"’</span>
<span class="r-in"><span><span class="fu">counts</span><span class="op">(</span><span class="va">factRObject</span>, <span class="st">"Ptbp1"</span>, set <span class="op">=</span> <span class="st">"gene"</span>, slot <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span> <span class="co"># get gene-level count</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in (function (classes, fdef, mtable) {    methods &lt;- .findInheritedMethods(classes, fdef, mtable)    if (length(methods) == 1L)         return(methods[[1L]])    else if (length(methods) == 0L) {        cnames &lt;- paste0("\"", vapply(classes, as.character,             ""), "\"", collapse = ", ")        stop(gettextf("unable to find an inherited method for function %s for signature %s",             sQuote(fdef@generic), sQuote(cnames)), domain = NA)    }    else stop("Internal error in finding inherited methods; didn't return a unique method",         domain = NA)})(list(structure("factR", package = "factR2")), new("standardGeneric",     .Data = function (object, ...)     standardGeneric("counts"), generic = structure("counts", package = "BiocGenerics"),     package = "BiocGenerics", group = list(), valueClass = character(0),     signature = "object", default = NULL, skeleton = (function (object,         ...)     stop(gettextf("invalid call in method dispatch to '%s' (no default method)",         "counts"), domain = NA))(object, ...)), &lt;environment&gt;):</span> unable to find an inherited method for function ‘counts’ for signature ‘"factR"’</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## access samples metadata</span></span></span>
<span class="r-in"><span><span class="fu">samples</span><span class="op">(</span><span class="va">factRObject</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            proj.ident</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P14_M5_1 factRProject</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P14_M5_2 factRProject</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P14_M5_3 factRProject</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P14_M6_1 factRProject</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P14_M6_2 factRProject</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P14_M6_3 factRProject</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P28_M1_1 factRProject</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P28_M1_2 factRProject</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P28_M1_3 factRProject</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P28_M2_1 factRProject</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P28_M2_2 factRProject</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> P28_M2_3 factRProject</span>
<span class="r-in"><span><span class="fu">ident</span><span class="op">(</span><span class="va">factRObject</span><span class="op">)</span>   <span class="co"># prints out the current identity</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "proj.ident"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## run correlation between gene expression and exon inclusion</span></span></span>
<span class="r-in"><span><span class="va">factRObject</span> <span class="op">&lt;-</span> <span class="fu">testGeneCorr</span><span class="op">(</span><span class="va">factRObject</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in dplyr::select(., AS_id, gene_id, ASNMDtype):</span> Can't subset columns that don't exist.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Column `ASNMDtype` doesn't exist.</span>
<span class="r-in"><span><span class="fu">ase</span><span class="op">(</span><span class="va">factRObject</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ </span><span style="color: #0000BB; font-style: italic;">Set `show_more to TRUE to show more info`</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 48,197 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    AS_id   gene_id               gene_name coord       AStype strand width novel</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> AS01129 ENSMUSG00000033813.16 Tcea1     chr1:49570… AD     +        231 yes  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> AS01642 ENSMUSG00000025907.15 Rb1cc1    chr1:62851… AD     +       <span style="text-decoration: underline;">1</span>104 yes  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> AS01647 ENSMUSG00000025907.15 Rb1cc1    chr1:62929… AF     +        499 yes  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> AS01649 ENSMUSG00000025907.15 Rb1cc1    chr1:63043… RI     +        141 yes  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> AS01650 ENSMUSG00000025907.15 Rb1cc1    chr1:63152… RI     +        178 yes  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> AS01651 ENSMUSG00000025907.15 Rb1cc1    chr1:63155… RI     +         80 yes  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> AS01652 ENSMUSG00000025907.15 Rb1cc1    chr1:63198… RI     +         87 yes  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> AS01657 ENSMUSG00000025907.15 Rb1cc1    chr1:63313… AD     +        749 yes  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> AS01658 ENSMUSG00000025907.15 Rb1cc1    chr1:63331… RI     +         81 yes  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> AS01659 ENSMUSG00000025907.15 Rb1cc1    chr1:63334… AD     +        173 yes  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 48,187 more rows</span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

