<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Display factR2 object data — factR-meta • factR2</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Display factR2 object data — factR-meta"><meta property="og:description" content="A factRObject-class contains different types of data at the gene, transcript,
 alternative splicing (AS) and protein domain levels. The functions below are
 designed to display specific contents of a factRObject."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">factR2</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/factR2.html">
    <span class="fas fa-book-open"></span>
     
    Using factR2
  </a>
</li>
<li>
  <a href="../news/index.html">
    <span class="fa fa-newspaper"></span>
     
    News
  </a>
</li>
<li>
  <a href="../reference/index.html">
    <span class="fa fa-code"></span>
     
    Functions
  </a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="..//index.html">
    <span class="fas fa-globe"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/f-hamidlab/factR2" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/FurshamH" class="external-link">
    <span class="fab fa-twitter fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Display factR2 object data</h1>
    
    <div class="hidden name"><code>factR-meta.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>A factRObject-class contains different types of data at the gene, transcript,
 alternative splicing (AS) and protein domain levels. The functions below are
 designed to display specific contents of a factRObject.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="co"># S4 method for factR</span></span>
<span><span class="fu">granges</span><span class="op">(</span><span class="va">object</span>, <span class="va">...</span>, set <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for factR</span></span>
<span><span class="fu">activeSet</span><span class="op">(</span><span class="va">object</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for factR</span></span>
<span><span class="fu">activeSet</span><span class="op">(</span><span class="va">object</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">value</span></span>
<span></span>
<span><span class="co"># S4 method for factR</span></span>
<span><span class="fu">listSets</span><span class="op">(</span><span class="va">object</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for factR</span></span>
<span><span class="fu">features</span><span class="op">(</span><span class="va">object</span>, <span class="va">...</span>, set <span class="op">=</span> <span class="cn">NULL</span>, show_more <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for factR</span></span>
<span><span class="fu">genes</span><span class="op">(</span><span class="va">object</span>, <span class="va">...</span>, show_more <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for factR</span></span>
<span><span class="fu">gns</span><span class="op">(</span><span class="va">object</span>, <span class="va">...</span>, show_more <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for factR</span></span>
<span><span class="fu">transcripts</span><span class="op">(</span><span class="va">object</span>, <span class="va">...</span>, show_more <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for factR</span></span>
<span><span class="fu">txs</span><span class="op">(</span><span class="va">object</span>, <span class="va">...</span>, show_more <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for factR</span></span>
<span><span class="fu">ase</span><span class="op">(</span><span class="va">object</span>, <span class="va">...</span>, show_more <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for factR</span></span>
<span><span class="fu">domains</span><span class="op">(</span><span class="va">object</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>object</dt>
<dd><p>factRObject</p></dd>


<dt>...</dt>
<dd><p>One or more features to display. Can be the following:</p><ul><li><p>gene_id: ID of gene to plot</p></li>
<li><p>gene_name: Name of gene to plot</p></li>
<li><p>transcript_id: ID of transcript to plot</p></li>
</ul></dd>


<dt>set</dt>
<dd><p>Set metadata to display. Can be "gene", "transcript" or "AS".</p></dd>


<dt>value</dt>
<dd><p>Character value of one of the following: "gene", "transcript" or "AS"</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<ul><li><p>`granges`: GenomicRanges object of selected features</p></li>
<li><p>`activeSet` and `listSets`: Character value/vector</p></li>
<li><p>All other functions: Tibble dataframe containing metadata of selected features</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">### Load sample factRObject</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"factRsample"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Prints out activeSet</span></span></span>
<span class="r-in"><span><span class="fu">activeSet</span><span class="op">(</span><span class="va">factRsample</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "AS"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Change activeSet</span></span></span>
<span class="r-in"><span><span class="fu">activeSet</span><span class="op">(</span><span class="va">factRsample</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"transcript"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Returns coordinates and metadata of features as a GenomicRanges object</span></span></span>
<span class="r-in"><span><span class="fu">granges</span><span class="op">(</span><span class="va">factRsample</span><span class="op">)</span>   <span class="co"># from activeSet</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in (function (classes, fdef, mtable) {    methods &lt;- .findInheritedMethods(classes, fdef, mtable)    if (length(methods) == 1L)         return(methods[[1L]])    else if (length(methods) == 0L) {        cnames &lt;- paste0("\"", vapply(classes, as.character,             ""), "\"", collapse = ", ")        stop(gettextf("unable to find an inherited method for function %s for signature %s",             sQuote(fdef@generic), sQuote(cnames)), domain = NA)    }    else stop("Internal error in finding inherited methods; didn't return a unique method",         domain = NA)})(list(structure("factR", package = "factR2")), new("standardGeneric",     .Data = function (x, use.names = TRUE, use.mcols = FALSE,         ...)     standardGeneric("granges"), generic = structure("granges", package = "GenomicRanges"),     package = "GenomicRanges", group = list(), valueClass = character(0),     signature = "x", default = NULL, skeleton = (function (x,         use.names = TRUE, use.mcols = FALSE, ...)     stop(gettextf("invalid call in method dispatch to '%s' (no default method)",         "granges"), domain = NA))(x, use.names, use.mcols, ...)),     &lt;environment&gt;):</span> unable to find an inherited method for function ‘granges’ for signature ‘"factR"’</span>
<span class="r-in"><span><span class="fu">granges</span><span class="op">(</span><span class="va">factRsample</span>, <span class="st">"Dab2"</span><span class="op">)</span>   <span class="co"># specific features</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in (function (classes, fdef, mtable) {    methods &lt;- .findInheritedMethods(classes, fdef, mtable)    if (length(methods) == 1L)         return(methods[[1L]])    else if (length(methods) == 0L) {        cnames &lt;- paste0("\"", vapply(classes, as.character,             ""), "\"", collapse = ", ")        stop(gettextf("unable to find an inherited method for function %s for signature %s",             sQuote(fdef@generic), sQuote(cnames)), domain = NA)    }    else stop("Internal error in finding inherited methods; didn't return a unique method",         domain = NA)})(list(structure("factR", package = "factR2")), new("standardGeneric",     .Data = function (x, use.names = TRUE, use.mcols = FALSE,         ...)     standardGeneric("granges"), generic = structure("granges", package = "GenomicRanges"),     package = "GenomicRanges", group = list(), valueClass = character(0),     signature = "x", default = NULL, skeleton = (function (x,         use.names = TRUE, use.mcols = FALSE, ...)     stop(gettextf("invalid call in method dispatch to '%s' (no default method)",         "granges"), domain = NA))(x, use.names, use.mcols, ...)),     &lt;environment&gt;):</span> unable to find an inherited method for function ‘granges’ for signature ‘"factR"’</span>
<span class="r-in"><span><span class="fu">granges</span><span class="op">(</span><span class="va">factRsample</span>, <span class="st">"Dab2"</span>, set <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span>   <span class="co"># specific features from different Set</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in (function (classes, fdef, mtable) {    methods &lt;- .findInheritedMethods(classes, fdef, mtable)    if (length(methods) == 1L)         return(methods[[1L]])    else if (length(methods) == 0L) {        cnames &lt;- paste0("\"", vapply(classes, as.character,             ""), "\"", collapse = ", ")        stop(gettextf("unable to find an inherited method for function %s for signature %s",             sQuote(fdef@generic), sQuote(cnames)), domain = NA)    }    else stop("Internal error in finding inherited methods; didn't return a unique method",         domain = NA)})(list(structure("factR", package = "factR2")), new("standardGeneric",     .Data = function (x, use.names = TRUE, use.mcols = FALSE,         ...)     standardGeneric("granges"), generic = structure("granges", package = "GenomicRanges"),     package = "GenomicRanges", group = list(), valueClass = character(0),     signature = "x", default = NULL, skeleton = (function (x,         use.names = TRUE, use.mcols = FALSE, ...)     stop(gettextf("invalid call in method dispatch to '%s' (no default method)",         "granges"), domain = NA))(x, use.names, use.mcols, ...)),     &lt;environment&gt;):</span> unable to find an inherited method for function ‘granges’ for signature ‘"factR"’</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Returns metadata of features</span></span></span>
<span class="r-in"><span><span class="fu">features</span><span class="op">(</span><span class="va">factRsample</span><span class="op">)</span>   <span class="co"># from activeSet</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ </span><span style="color: #0000BB; font-style: italic;">Set `show_more to TRUE to show more info`</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 154 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    transcript_id         gene_id        gene_name strand width novel cds   nmd  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> ENSMUST00000022746.12 ENSMUSG000000… Osmr      -       <span style="text-decoration: underline;">5</span>491 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> ENSMUST00000175862.7  ENSMUSG000000… Osmr      -       <span style="text-decoration: underline;">2</span>837 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> ENSMUST00000176554.1  ENSMUSG000000… Osmr      -        908 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> ENSMUST00000176826.1  ENSMUSG000000… Osmr      -       <span style="text-decoration: underline;">4</span>028 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> ENSMUST00000177263.1  ENSMUSG000000… Osmr      -       <span style="text-decoration: underline;">6</span>160 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> ENSMUST00000177478.1  ENSMUSG000000… Osmr      -       <span style="text-decoration: underline;">1</span>727 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> MSTRG.14573.1         ENSMUSG000000… Osmr      *        877 yes   no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> ENSMUST00000090461.11 ENSMUSG000000… Fyb       +       <span style="text-decoration: underline;">2</span>678 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> ENSMUST00000159604.1  ENSMUSG000000… Fyb       +        771 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> ENSMUST00000159714.1  ENSMUSG000000… Fyb       +       <span style="text-decoration: underline;">2</span>170 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 144 more rows</span></span>
<span class="r-in"><span><span class="fu">features</span><span class="op">(</span><span class="va">factRsample</span>, <span class="st">"Dab2"</span><span class="op">)</span>   <span class="co"># specific features</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ </span><span style="color: #0000BB; font-style: italic;">Set `show_more to TRUE to show more info`</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 16 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    transcript_id         gene_id        gene_name strand width novel cds   nmd  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> ENSMUST00000078019.12 ENSMUSG000000… Dab2      +       <span style="text-decoration: underline;">3</span>793 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> ENSMUST00000080880.11 ENSMUSG000000… Dab2      +       <span style="text-decoration: underline;">4</span>670 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> ENSMUST00000110663.8  ENSMUSG000000… Dab2      +       <span style="text-decoration: underline;">3</span>838 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> ENSMUST00000110664.8  ENSMUSG000000… Dab2      +       <span style="text-decoration: underline;">4</span>457 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> ENSMUST00000159046.1  ENSMUSG000000… Dab2      +        438 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> ENSMUST00000159490.7  ENSMUSG000000… Dab2      +        444 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> ENSMUST00000159552.1  ENSMUSG000000… Dab2      +        922 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> ENSMUST00000160134.7  ENSMUSG000000… Dab2      +       <span style="text-decoration: underline;">1</span>460 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> ENSMUST00000161040.7  ENSMUSG000000… Dab2      +       <span style="text-decoration: underline;">1</span>184 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> ENSMUST00000161558.1  ENSMUSG000000… Dab2      +       <span style="text-decoration: underline;">2</span>119 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">11</span> ENSMUST00000161812.7  ENSMUSG000000… Dab2      +       <span style="text-decoration: underline;">3</span>774 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">12</span> ENSMUST00000162094.2  ENSMUSG000000… Dab2      +        737 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">13</span> ENSMUST00000162140.1  ENSMUSG000000… Dab2      +       <span style="text-decoration: underline;">1</span>406 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">14</span> ENSMUST00000163082.1  ENSMUSG000000… Dab2      +       <span style="text-decoration: underline;">3</span>839 no    no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">15</span> MSTRG.14564.1         ENSMUSG000000… Dab2      *        213 yes   no    no   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">16</span> MSTRG.14566.1         ENSMUSG000000… Dab2      *        202 yes   no    no   </span>
<span class="r-in"><span><span class="fu">features</span><span class="op">(</span><span class="va">factRsample</span>, <span class="st">"Dab2"</span>, set <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span>   <span class="co"># specific features from different Set</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ </span><span style="color: #0000BB; font-style: italic;">Set `show_more to TRUE to show more info`</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1 × 5</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   gene_id               gene_name strand  width match_level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> ENSMUSG00000022150.16 Dab2      +      <span style="text-decoration: underline;">140</span>925           4</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### This is the same as:</span></span></span>
<span class="r-in"><span><span class="fu">genes</span><span class="op">(</span><span class="va">factRsample</span>, <span class="st">"Dab2"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ </span><span style="color: #0000BB; font-style: italic;">Set `show_more to TRUE to show more info`</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1 × 5</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   gene_id               gene_name strand  width match_level</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> ENSMUSG00000022150.16 Dab2      +      <span style="text-decoration: underline;">140</span>925           4</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## To return protein-coding domains, the protein-coding domains need to be predicted first:</span></span></span>
<span class="r-in"><span><span class="va">factRsample</span> <span class="op">&lt;-</span> <span class="fu"><a href="buildCDS.html">buildCDS</a></span><span class="op">(</span><span class="va">factRsample</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">factRsample</span> <span class="op">&lt;-</span> <span class="fu">getAAsequence</span><span class="op">(</span><span class="va">factRsample</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">factRsample</span> <span class="op">&lt;-</span> <span class="fu">predictDomains</span><span class="op">(</span><span class="va">factRsample</span>, <span class="st">"Dab2"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ </span><span style="color: #0000BB; font-style: italic;">Set `show_more to TRUE to show more info`</span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Skipped 6 non-coding transcripts</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Then, the domains of the selected gene can be printed as such:</span></span></span>
<span class="r-in"><span><span class="fu">domains</span><span class="op">(</span><span class="va">factRsample</span>, <span class="st">"Dab2"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 8 × 6</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   type   description    eval    begin   end transcript_id        </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> DOMAIN PH domain-like 2.7e-42    33   180 ENSMUST00000078019.12</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> DOMAIN PH domain-like 4.9e-42    33   180 ENSMUST00000080880.11</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">3</span> DOMAIN PH domain-like 2.7e-42    33   180 ENSMUST00000110663.8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">4</span> DOMAIN PH domain-like 4.7e-42    33   180 ENSMUST00000110664.8 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">5</span> DOMAIN PH domain-like 3.9e-43    33   180 ENSMUST00000160134.7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">6</span> DOMAIN PH domain-like 3.8e-43    33   180 ENSMUST00000161040.7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">7</span> DOMAIN PH domain-like 2.5e-42    33   180 ENSMUST00000161812.7 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">8</span> DOMAIN PH domain-like 2.6e-21    35   110 ENSMUST00000162094.2 </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## All outputs can be assigned to a variable and manipulated further using other functions:</span></span></span>
<span class="r-in"><span><span class="fu">ase</span><span class="op">(</span><span class="va">factRsample</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">AStype</span> <span class="op">==</span> <span class="st">"CE"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in ase(factRsample) %&gt;% dplyr::filter(AStype == "CE"):</span> could not find function "%&gt;%"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

